test/unit_tests/test_braket_device.py

```diff
@@ -625,23 +652,93 @@ def f(thetas, **kwargs):
     assert set(samples) == {0, 1}
 
 
+@pytest.mark.xfail(raises=AttributeError)
+def test_none_device():
+    """Tests that an error is thrown when device is not given"""
+    dev = DummyLocalQubitDevice(wires=2, device=None, shots=1000)
+
+
 @pytest.mark.xfail(raises=NotImplementedError)
 def test_run_task_unimplemented():
     """Tests that an error is thrown when _run_task is not implemented"""
-    dev = DummyLocalQubitDevice(wires=2, device=None, shots=1000)
+    dummy = DummyCircuitSimulator()
+    dev = DummyLocalQubitDevice(wires=2, device=dummy, shots=1000)
```

Comment:
Might be worth looking into whether this causing `NotImplementedError` to be thrown at a different point from expected, causing coverage to drop.

Reply:
The device not having properties results in the `NotImplementedError` not being thrown